### ISSUE TITLE
question - button spacing

### ISSUE THREAD

**Issue by pablocar80** [picoe/Eto#1384]
Hello! I created buttons with images, is there any way to add some **spacing between the image and the borders of the button**? Or alternatively, not show the borders? I tried making the buttons larger, however, this makes the images larger too.

Here's how the buttons look:

![image](https://user-images.githubusercontent.com/36197603/57043441-1d280180-6c2d-11e9-8580-04b2c4fd8364.png)

**Comment by SlowLogicBoy** [comment]
try adding empty label where you want spaces to be. `new Label()`

**Comment by pablocar80** [comment]
A label inside a button you mean? I need 2 pixels more of spacing on the top, left and bottom of the image, it’s too close to the button’s borders.

**Comment by pablocar80** [comment]
Something like this:

![image](https://user-images.githubusercontent.com/36197603/57078624-32943e80-6cb4-11e9-8b5c-efb16b737bff.png)

**Comment by cwensley** [comment]
@pablocar80 what platform are you noticing this on?  I don't recognize that theme, perhaps this is GTK?  Also, have you tried Eto 2.5?

**Comment by pablocar80** [comment]
It's WPF on Eto 2.5.0-beta.1. Here's a full screenshot:

![image](https://user-images.githubusercontent.com/36197603/69498111-e73cba00-0ea9-11ea-8267-d9e984ac6dbe.png)

As a reference, this is how the screen used to look on Winforms before migrating to Eto:

![image](https://user-images.githubusercontent.com/36197603/69498122-089da600-0eaa-11ea-89d1-c75c458895f2.png)

I'm using Windows 10 with a 4K monitor, VS2019, all latest updates.

This issue is probably closely related to the other issue of having default themes for Windows, so that things look nice out of the box. I believe that's the main reason so many folks choose Avalonia.

**Comment by SlowLogicBoy** [comment]
My Solution is to use Styling:

```csharp
using System;
using System.Reflection;
using Eto.Drawing;
using Eto.Forms;
using Eto.Wpf.Forms.Controls;

namespace Tests
{
    //In Wpf executable
    class Program
    {
        [STAThread]
        static void Main(string[] args)
        {
            //Set native Style
            Eto.Style.Add<Button>("MyStyle", (btn) => {
                //You'll need to reference "PresentationFramework"
                //Set Border Thickness
                var wpfButton = btn.ControlObject as System.Windows.Controls.Button;
                wpfButton.BorderThickness = new System.Windows.Thickness(0);

                //Set Spacing through reflection magic
                var buttonHandler = btn.Handler as ButtonHandler;
                var wpfLabelField = buttonHandler.GetType().GetField("label", BindingFlags.NonPublic | BindingFlags.Instance);
                var wpfLabel = wpfLabelField.GetValue(buttonHandler) as System.Windows.Controls.Label;
                wpfLabel.HorizontalAlignment = System.Windows.HorizontalAlignment.Left;
            });
            new Application(new Eto.Wpf.Platform()).Run(new MainForm());
        }
    }
    //In Eto Library
    class MainForm : Form
    {
        public MainForm()
        {
            Padding = 3;
            var button = new Button{
                Style = "MyStyle", //Set Style
                Image = new Bitmap(@"Path/To/My/Image.png"),
                Text = "Asterisk"
            };
            Content = new StackLayout()
            {
                Items = 
                {
                    button
                }
            };
        }
    }
}

```
Before (Without "MyStyle" Style):
![image](https://user-images.githubusercontent.com/3121637/69519162-fda64c80-0f61-11ea-9e74-818a1a5b86cf.png)

After (With "MyStyle" Style):
![image](https://user-images.githubusercontent.com/3121637/69521071-d30ac280-0f66-11ea-8b9d-2088500d3d66.png)


This is sketchy and uses reflection but it does the job.

[Wpf button](https://github.com/picoe/Eto/blob/develop/src/Eto.Wpf/Forms/Controls/ButtonHandler.cs#L76-L85)

**Comment by pablocar80** [comment]
Thank you for the example @SlowLogicBoy.

This kind of reinforces my point. IMHO beginners shouldn't need to resort to advanced settings and/or hack the platform using reflection to make it look good.

**Comment by cwensley** [comment]
Ok the spacing of the image should now be the same as winforms (2px) in the develop branch.  I've also exposed the LabelPart and ImagePart as properties of the ButtonHandler so you can easily style it without hacky reflection like so:

```c#
Style.Add<Eto.Wpf.Forms.Controls.ButtonHandler>(null, h =>
{
	h.Control.BorderThickness = new System.Windows.Thickness(0);
	h.LabelPart.HorizontalAlignment = System.Windows.HorizontalAlignment.Left; // change text alignment
	// h.Control.Padding = new System.Windows.Thickness(4); // change padding in button if desired
});
```

This fix will be included with 2.5.1 when released.
